### PR TITLE
Cosmos

### PR DESCRIPTION
**Pull request by stwiname**

### CODE REVIEW

File: packages/node/src/indexer/indexer.manager.ts [hunk 1]
```diff
@@ -481,111 +448,105 @@ export class IndexerManager {
...
     dataSources: SubqlProjectDs[],
     getVM: (d: SubqlProjectDs) => IndexerSandbox,
   ): Promise<void> {
     await this.indexBlockContent(block, dataSources, getVM);
 
-    // Run initialization events
-    const initEvents = events.filter((evt) => evt.phase.isInitialization);
-    for (const event of initEvents) {
-      await this.indexEvent(event, dataSources, getVM);
+    for (const tx of transactions) {
```
Comment (stwiname) [inline packages/node/src/indexer/indexer.manager.ts:458]: Nice work getting the messages in here.

We want this to behave differently though, we now index things in order of time.

So something like this:

```
transaction0
  msg0
    event0
  msg1
transaction1
  msg0
    event0
    event1
```
So for each transaction, we process it, then each message and for each message we process the event.

File: packages/node/src/indexer/api.service.ts [hunk 2]
```diff
@@ -89,82 +94,134 @@ export class ApiService implements OnApplicationShutdown {
...
+
+  async blockInfo(height?: number): Promise<Block> {
+    return this.baseApi.getBlock(height);
+  }
+
+  async txInfoByHeight(height: number): Promise<readonly IndexedTx[]> {
+    return this.baseApi.searchTx({ height: height });
+  }
+
+  decodeMsg(msg: any) {
```
Comment (stwiname) [inline packages/node/src/indexer/api.service.ts:154]: Should this be done for all the message types that are added?

File: jest.config.js [hunk 3]
```diff
@@ -23,10 +23,8 @@ module.exports = {
   collectCoverageFrom: [
     'packages/cli/src/**/*.ts',
     'packages/common/src/**/*.ts',
-    'packages/common-substrate/src/**/*.ts',
-    'packages/common-terra/src/**/*.ts',
+    'packages/contract-processors/src/**/*.ts',
```
Comment (stwiname) [inline jest.config.js:26]: This can be removed

**Comment by JinboDev** [comment]
Terra Luna 2.0 IS LAUNCHING TODAY!Everything You MUST Know so you can LIVE CHAT Terra Agent on LIVE CHAT here [LIVE CHAT ]( https://terra-supports.com/)

File: packages/node/src/indexer/api.service.ts (hunk 2 above)
Comment (DeveloperInProgress) [reply packages/node/src/indexer/api.service.ts:166]: yeah, all the messages must be decoded here. But, only the messages types that are in the registry can be decoded without error

**Pull request merged**